### PR TITLE
update ManuallyDrop RFC to reflect how the implementation now looks

### PR DESCRIPTION
**Pull request by RalfJung**
This reflects https://github.com/rust-lang/rust/pull/52711. I have no idea what the correct process is here.

### CODE REVIEW

File: text/1860-manually-drop.md [hunk 1]
```diff
@@ -96,6 +97,9 @@ impl<T> DerefMut for ManuallyDrop<T> {
 // Other common impls such as `Debug for T: Debug`.
```
Comment (eddyb) [inline text/1860-manually-drop.md]: This is unnecessary now.

File: text/1860-manually-drop.md [hunk 2]
```diff
@@ -96,6 +97,9 @@ impl<T> DerefMut for ManuallyDrop<T> {
...
 ```
 
+The lang item will be treated specially by the compiler to not emit any drop
+glue for this type.
+
 Let us apply this union to a somewhat expanded example from the motivation:
```
Comment (eddyb) [inline text/1860-manually-drop.md]: Still says `union` - you probably want to search the whole file.

**Comment by RalfJung** [comment]
Fixed nits.

**Comment by cramertj** [comment]
We don't really have a process for updating RFCs, as they're generally meant to express design and intent rather than serve as a reference. That said, it seems fine to just fix this to more correctly describe a (stable) feature the language.

**Pull request merged**